core/policy/src/main/java/orca/policy/core/ServiceManagerTicketReviewPolicy.java

```diff
@@ -0,0 +1,188 @@
+package orca.policy.core;
+
+import orca.shirako.api.IReservation;
+import orca.shirako.common.SliceID;
+import orca.shirako.kernel.IKernelSlice;
+import orca.shirako.util.ReservationSet;
+import orca.util.persistence.NotPersistent;
+
+import java.util.*;
+
+import static orca.manage.OrcaConstants.*;
+
+/**
+ * This implementation of a Service Manager policy is almost identical to the parent
+ * ServiceManagerSimplePolicy.
+ *
+ * The only real difference is that it addresses the following issue:
+ * https://github.com/RENCI-NRIG/orca5/issues/88
+ * Tickets should not be redeemed if any reservations are currently Failed or Nascent.
+ *
+ */
+public class ServiceManagerTicketReviewPolicy extends ServiceManagerSimplePolicy {
+
+    @NotPersistent
+    protected ReservationSet pendingRedeem;
+
+    public ServiceManagerTicketReviewPolicy(){
+        super();
+        pendingRedeem = new ReservationSet();
+    }
+
+    /**
+     * Check to make sure all reservations are Ticketed (not Failed or Nascent)
+     * before calling the parent method.
+     *
+     * @throws Exception
+     */
+    @Override
+    protected void checkPending() throws Exception {
+
+        // add all of our pendingRedeem, so they can be checked
+        for (IReservation reservation : pendingRedeem) {
+            calendar.addPending(reservation);
+        }
+
+        // get set of reservations that need to be redeemed
+        ReservationSet myPending = calendar.getPending();
+
+        // keep track of status of the slice containing each reservation
+        Map<SliceID, Integer> sliceStatusMap = new HashMap<>();
```

Comment:
I don't understand the need for the map. It seems everything happens in a loop once per reservation on the pending ReservationSet, within which SliceID by which the map is referenced doesn't change...